Classes/Utility/SearchQueryTokenizer.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  SearchQueryTokenizer.swift
+//  Freetime
+//
+//  Created by Weyert de Boer on 09/10/2017.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+struct SearchQueryParseResults {
+    let tokens: [Token]
+    let leftover: String
+}
+
+enum TokenAction: String {
+    case status = "is"
+    case user = "user"
+}
+
+typealias Token = (action: TokenAction, value: String)
+
+/// Tokenizes a search query and returns an instance of ParseResult which consists
+/// of all the found "search tokens" such as `is:closed` and separately returns
+/// the text of the search query which isn't part of any search token
+func tokenizeSearchQuery(_ query: String) -> SearchQueryParseResults {
```

Comment:
Genuinely looking forward to this functionality! Such a drag scrolling through tonnes of closed issues to get to old open ones! Almost feel we should mimmick GitHub and default search to `is:open` 🤔 

Great work overall!

Reply:
Yes, once this commit is in I will make a change so that you can search the inbox in a style of `is:pr` / `is:issue`, `type:pr` and `type:issue`.

Reply:
@Sherlouk ya, I like the idea of defaulting to open. Much easier. I think we'll want to extend this search bar eventually. Maybe use GH's `is:open` stuff?

Reply:
`is:open` is already supported by this PR! 😅 That was my whole feature request!